### PR TITLE
CI - add test for HPA enabled

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Description
Basic test (we only verify `helm upgrade --install`) for our HPA setup. 

We are adding it to catch future regressions like https://github.com/PostHog/charts-clickhouse/pull/254 

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## How has this been tested?
CI ✅ 

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ci/kubetest/test_clickhouse_persistence_disabled.py [hunk 1]
```diff
@@ -22,7 +22,11 @@ def setup(kube):
     wait_for_pods_to_be_ready(kube)
 
 
-def test_volume_claim(setup, kube):
+def test_helm_install(setup, kube):
```
Comment (hazzadous) [inline ci/kubetest/test_clickhouse_persistence_disabled.py:25]: This looks a bit funny with the actual bit being tested in a fixture, could we instead run the commands within the test function?

I'm becoming more of the opinion that pytest fixture are cute, but not obvious

File: ci/kubetest/test_clickhouse_persistence_disabled.py [hunk 2]
```diff
@@ -15,14 +15,13 @@
...
-@pytest.fixture
-def setup(kube):
+def test_helm_install(setup, kube):
     cleanup_k8s()
     helm_install(HELM_INSTALL_CMD)
     wait_for_pods_to_be_ready(kube)
 
 
-def test_volume_claim(setup, kube):
+def test_volume_claim(kube):
```
Comment (hazzadous) [inline ci/kubetest/test_clickhouse_persistence_disabled.py:24]: Does this mean we rely on `test_helm_install` having run prior to this one?

File: ci/kubetest/test_clickhouse_persistence_disabled.py [hunk 3]
```diff
@@ -15,14 +15,13 @@
 """
 
 
-@pytest.fixture
-def setup(kube):
+def test_helm_install(setup, kube):
```
Comment (hazzadous) [inline ci/kubetest/test_clickhouse_persistence_disabled.py:18]: Where is the `setup` fixture defined?

File: ci/kubetest/test_clickhouse_persistence_disabled.py (hunk 2 above)
Comment (guidoiaquinti) [reply ci/kubetest/test_clickhouse_persistence_disabled.py:24]: Yes

Comment (hazzadous) [reply ci/kubetest/test_clickhouse_persistence_disabled.py:24]: I think we should just drop the non-hpa test changes and get this in, is that possible?

I think test interdependence is something we should avoid, as will make it hard to e.g. like you said introduce parallelism among other things

**Review by hazzadous — APPROVED** [review]

**Pull request merged**